native/src/components/TtsContainer.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { createContext, ReactElement, useCallback, useEffect, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { AppState } from 'react-native'
+import Tts from 'react-native-tts'
+
+import { truncate } from 'shared/utils/getExcerpt'
+
+import { AppContext } from '../contexts/AppContextProvider'
+import { reportError } from '../utils/sentry'
+import TtsPlayer from './TtsPlayer'
+
+const MAX_TITLE_DISPLAY_CHARS = 20
+
+export type TtsContextType = {
+  enabled: boolean
+  setEnabled: (enable: boolean) => void
+  visible: boolean
+  setVisible: (visible: boolean) => void
+  sentences: string[] | null
+  setSentences: (sentences: string[]) => void
+}
+
+export const ttsContext = createContext<TtsContextType>({
+  enabled: false,
+  setEnabled: () => undefined,
+  visible: false,
+  setVisible: () => undefined,
+  sentences: [],
+  setSentences: () => undefined,
+})
+
+type TtsContainerProps = {
+  children: ReactElement
+}
+
+const TtsContainer = ({ children }: TtsContainerProps): ReactElement => {
+  const { languageCode } = React.useContext(AppContext)
+  const { t } = useTranslation('layout')
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [sentenceIndex, setSentenceIndex] = useState(0)
+  const [visible, setVisible] = useState(false)
+  const [enabled, setEnabled] = useState(false)
```

Comment:
:x: There is no state needed for this as it does need to be manually set (enabled is just dependent on other states/constant values). See comments https://github.com/digitalfabrik/integreat-app/pull/2950#discussion_r1868047660 and https://github.com/digitalfabrik/integreat-app/pull/2950#discussion_r1868056687
This should be done here and only `enabled` should be passed down in the context and exposed by `useTtsPlayer`.